core_impl/src/main/java/io/opencensus/trace/StartEndHandlerImpl.java

```diff
@@ -14,51 +14,103 @@
 package io.opencensus.trace;
 
 import io.opencensus.internal.EventQueue;
+import io.opencensus.trace.Span.Options;
 import io.opencensus.trace.SpanImpl.StartEndHandler;
+import io.opencensus.trace.export.ActiveSpansExporterImpl;
+import io.opencensus.trace.export.SampledSpanStore;
 import io.opencensus.trace.export.SpanData;
 import io.opencensus.trace.export.SpanExporterImpl;
+import javax.annotation.Nullable;
 
 /**
  * Uses the provided {@link EventQueue} to defer processing/exporting of the {@link SpanData} to
  * avoid impacting the critical path.
  */
 public final class StartEndHandlerImpl implements StartEndHandler {
+  private final SpanExporterImpl spanExporter;
+  private final ActiveSpansExporterImpl activeSpansExporter;
+  private final SampledSpanStore sampledSpanStore;
   private final EventQueue eventQueue;
-  private final SpanExporterImpl sampledSpansServiceExporter;
+  // true if any of (activeSpansExporter OR sampledSpanStore) are different than null, which
+  // means the spans with RECORD_EVENTS should be enqueued in the queue.
+  private final boolean enqueEventForNonSampledSpans;
 
+  /**
+   * Constructs a new {@code StartEndHandlerImpl}.
+   *
+   * @param spanExporter the {@code SpanExporter} implementation.
+   * @param activeSpansExporter the {@code ActiveSpansExporter} implementation.
+   * @param sampledSpanStore the {@code SampledSpanStore} implementation.
+   * @param eventQueue the event queue where all the events are enqueued.
+   */
   public StartEndHandlerImpl(
-      SpanExporterImpl sampledSpansServiceExporter, EventQueue eventQueue) {
-    this.sampledSpansServiceExporter = sampledSpansServiceExporter;
+      SpanExporterImpl spanExporter,
+      @Nullable ActiveSpansExporterImpl activeSpansExporter,
```

Comment:
Forgot to add that code. On the lite version we do not want to support active and sampled.